scripts/components/CollectionEdit.js

```diff
@@ -1,26 +1,50 @@
 import React, { Component } from "react";
+import Form from "react-jsonschema-form";
 
 import CollectionForm from "./CollectionForm";
 import Spinner from "./Spinner";
 
 
+const deleteSchema = {
+  type: "string",
+  title: "Please enter the collection name to delete as a confirmation",
+};
+
+function DeleteForm({cid, onSubmit}) {
+  const validate = (formData, errors) => {
+    if (formData !== cid) {
+      errors.addError("The collection name does not match.");
+    }
+    return errors;
+  };
+  return (
+    <Form
+      schema={deleteSchema}
+      validate={validate}
```

Comment:
[Yes!](https://github.com/mozilla-services/react-jsonschema-form/pull/163)

Though there's no way to define custom validation for userland collections in the admin just yet. Here it's only used internally.